lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -135,28 +135,81 @@ def inline_metadata(catalog, checksum_type)
       file = resource.to_ral
 
       if file.recurse?
+        # memoize all resources that require/subscribe the parent file
+        direct_dependents = { :require => [], :subscribe => []}
+        catalog.resources.find_all do |other|
+          if other.ref != resource.ref
+            [:require, :subscribe].each do |param|
+              # other[param] can be nil, String or Array
+              target_refs = [other[param]].flatten.compact
+              target_refs.each do |target|
```

Comment:
N-1 resources x 2 metaparams x M target refs where M varies, but is probably 0-2 for most resources.